section4/exercises/burrito.rb

```diff
@@ -1,19 +1,48 @@
-# Add the following methods to this burrito class and 
+# Add the following methods to this burrito class and
 # call the methods below the class:
 # 1. add_topping
 # 2. remove_topping
 # 3. change_protein
 
 class Burrito
   attr_reader :protein, :base, :toppings
+
   def initialize(protein, base, toppings)
     @protein  = protein
     @base     = base
     @toppings = toppings
   end
+
+  def add_topping(new_topping)
+    @toppings << new_topping
+    p "You've added #{new_topping} to your burrito!"
+  end
+
+  def remove_topping(sad_topping)
+    if toppings.include?(sad_topping) == true
+      toppings.delete(sad_topping)
+      @toppings = toppings
+      p "You've removed #{sad_topping} from your burrito!"
+    else
+      p "You didn't have #{sad_topping} on your burrito!"
+    end
+  end
```

Comment:
✅ Nice! And excellent work putting in a guard statement to prevent toppings that aren't on the burrito from causing an error!